crates/data_transfer_objects/src/dto_attestation.rs

```diff
@@ -0,0 +1,187 @@
+use std::fmt;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use derive_more::Constructor;
+use serde::{Deserialize, Serialize};
+
+type Sha256Digest = [u8; 32];
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Debug, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+#[cfg_attr(
+    all(feature = "abi", not(target_arch = "wasm32")),
+    derive(schemars::JsonSchema)
+)]
+pub enum DtoAttestation {
```

Comment:
Nit: I prefer to skip the `Dto` prefix to the type here. The fact that this type lives in the DTO crate should be sufficient.

```suggestion
pub enum Attestation {
```